.github/actions/maven-deploy-file/action.yml

```diff
@@ -63,11 +63,16 @@ runs:
           echo "Existing settings.xml found"
           echo "Make sure to provide repository-id as input for choosing the right authentication credentials"
         fi
-    - shell: bash
+    - name: mvn deploy:deploy-file
+      shell: bash
       env:
         MAVEN_USERNAME: ${{ inputs.maven-username }}
         MAVEN_PASSWORD: ${{ inputs.maven-password }}
       run: |
+        OPTIONAL_ARGS=""
+        if [[ -n "${{ inputs.files }}" && -n "${{ inputs.classifiers }}" && -n "${{ inputs.types }}" ]]; then
+          OPTIONAL_ARGS="-Dfiles=${{ inputs.files }} -Dclassifiers=${{ inputs.classifiers }} -Dtypes=${{ inputs.types }}"
```

Comment:
I would actually quote all that between single quotes so each args can be double-quoted individually